lib/utilities.js

```diff
@@ -0,0 +1,79 @@
+var _ = require('lodash')
+
+function resultBlock (config, error, results) {
+  var data = _.clone(config.data)
+  data.error = data.error || error
+  data.results = data.results || results
+
+  return data
+}
+
+function mappingBlock (data, mapping, defaults, result) {
+  if (_.isString(mapping)) {
+    return _.get(data, mapping)
+  }
+
+  result = result || {}
+
+  result = _.transform(defaults, (target, value, key) => {
+    if (_.get(target, key, null) === null) {
+      _.set(target, key, value)
+    }
+  }, result)
+
+  result = _.transform(mapping, (target, value, key) => {
+    _.set(target, key, _.get(data, value))
+  }, result)
+
+  return result
+}
+
+function mappingArgs (data, mapping) {
+  var args = []
+
+  if (!_.isArray(mapping)) {
+    mapping = [mapping]
+  }
+
+  _.forEach(mapping, (value) => {
+    args.push(mappingBlock(data, value))
+  })
+
+  return args
+}
+
+function buildError (options) {
+  options = _.cloneDeep(options || {})
```

Comment:
I'm wondering how big options could get and if its necessary to clone the entire object? I could see buildError happening in looping logic and am wondering how big of a perf impact it could have.